core/services/ocr2/plugins/ccip/ccipexec/ocr2.go

```diff
@@ -476,31 +479,27 @@ func (r *ExecutionReportingPlugin) buildBatch(
 
 		expectedNonces[msg.Sender] = msg.Nonce + 1
 	}
+
 	return executableMessages
 }
 
-func getTokenData(ctx context.Context, lggr logger.Logger, msg internal.EVM2EVMOnRampCCIPSendRequestedWithMeta, tokenDataProviders map[common.Address]tokendata.Reader, skipTokenWithData bool) (tokenData [][]byte, err error) {
-	for _, token := range msg.TokenAmounts {
-		offchainTokenDataProvider, ok := tokenDataProviders[token.Token]
-		if !ok {
-			// No token data required
-			tokenData = append(tokenData, []byte{})
-			continue
-		}
-		if skipTokenWithData {
-			// If token data is required but should be skipped, exit without calling the API
-			return [][]byte{}, errors.New("token requiring data is flagged to be skipped")
-		}
-		lggr.Infow("Fetching token data", "token", token.Token.Hex())
-		tknData, err2 := offchainTokenDataProvider.ReadTokenData(ctx, msg)
-		if err2 != nil {
-			return [][]byte{}, err2
-		}
-
-		lggr.Infow("Token data retrieved", "token", token.Token.Hex())
-		tokenData = append(tokenData, tknData)
-	}
-	return tokenData, nil
+// getTokenDataWithCappedLatency gets the token data for the provided message.
+// Stops and returns an error if more than allowedWaitingTime is passed.
+func (r *ExecutionReportingPlugin) getTokenDataWithCappedLatency(
+	ctx context.Context,
+	msg internal.EVM2EVMOnRampCCIPSendRequestedWithMeta,
+	allowedWaitingTime time.Duration,
```

Comment:
`timeout` is better.

```suggestion
	timeout time.Duration,
```